vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -57,14 +58,14 @@ class Filter(VizroBaseModel):
     Args:
         type (Literal["filter"]): Defaults to `"filter"`.
         column (str): Column of `DataFrame` to filter.
-        targets (List[str]): Target component to be affected by filter. If none are given then target all components on
-            the page that use `column`.
+        targets (List[ModelID]): Target component to be affected by filter. If none are given then target all components
+            on the page that use `column`.
         selector (Optional[SelectorType]): See [SelectorType][vizro.models.types.SelectorType]. Defaults to `None`.
     """
 
     type: Literal["filter"] = "filter"
     column: str = Field(..., description="Column of DataFrame to filter.")
-    targets: List[str] = Field(
+    targets: List[ModelID] = Field(
```

Comment:
We discussed this before but not sure if you tried fixing the circular imports - I think it should be possible to do so in this case though. Either way, if `ModelID` appears in model docs then it should definitely live in `vizro.models.types` and not hidden away. Currently a user will see from the docs that the field requires a `ModelID` but not know what that is, which is actually worse than how things currently are where they know it's a string.

Let's do that as a separate PR though.

Reply:
Will create an issue for it